pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -297,6 +299,14 @@ cdef inline int64_t parse_timedelta_string(str ts) except? -1:
     if len(ts) == 0 or ts in nat_strings:
         return NPY_NAT
 
+    if re.search(r"^\d+\s?[M|Y|m|y]$", ts):
+        warnings.warn(
```

Comment:
you will instead want to do this in timespec_from_spec and not use a regex (they are slow compared to what is going on).

Reply:
Thanks for the tip, moved it.